address zsh for macOS Catalina for issue #653

moisbo opened the pull request:
Dear Carpentries,

Added a paragraph to address the changes made to Catalina. Using a support article from Apple for using zsh as it shows how to change the default shell on Catalina.

Cheers,
Moises

---

Thanks for contributing! If this contribution is for instructor training, please send an email to checkout@carpentries.org with a link to this contribution so we can record your progress. You’ve completed your contribution step for instructor checkout just by submitting this contribution.  

Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution. Although not all contributions can be incorporated into the lesson materials, we appreciate your time and effort to improve the curriculum.  If you have any questions about the lesson maintenance process or would like to volunteer your time as a contribution reviewer, please contact The Carpentries Team at team@carpentries.org.  

---

moisbo committed:
address zsh for macOS Catalina

ErinBecker commented:
Thank you @moisbo for your contribution! I know that bash and zsh are highly compatible. As far as I'm aware, there haven't been any issues with the current lesson working on Catalina. 

@gcapes @gdevenyi - as Maintainers of the SWC Unix Shell lesson, would you recommend that we tell learners running Cataline to switch their default shell to bash? Or does the lesson run smoothly on zsh?

brownsarahm commented:
We already added instructions for how to check and change the shell one paragraph below this suggestion.  

Perhaps it makes sense to move that paragraph up one, but I prefer the less specific (check if you have Bash) rather than mentioning Catalina by name at all. It makes for less maintainance, for example, if we don't mention it, then if the next version does or does not use bash as default doesn't require an update here. Also it avoids confusion for a user who might have catalina, but already changed their default at some other point in time.

fmichonneau commented:
I agree with Sarah.

fmichonneau commented:
@moisbo would you be willing to update this pull request to move up the instructions about checking and changing the shell and potentially add more information to make it clearer to everyone? Thanks!

moisbo commented:
Sure thing @fmichonneau

moisbo committed:
Merge branch 'gh-pages' into fix-653

moisbo committed:
removed catalina reference;
open terminal to access a shell -> see if default is bash -> change your 
dfault shell

moisbo commented:
Let me know if you agree with this one. Idea is: open terminal to access a shell -> see if default is bash -> Optional maybe: change your default shell

brownsarahm commented on _includes/swc/setup.html:
```diff
@@ -94,12 +94,16 @@ <h3>The Bash Shell</h3>
+            Bash by typing <code>bash</code>
+        </p>
+        <p>
+          To change your default shell open this <a href="https://support.apple.com/en-au/HT208050" rel="noopener">link</a>
+          and follow the instructions on "How to change your default shell".
```
```suggestion
          If you want to change your default shell, see <a href="https://support.apple.com/en-au/HT208050" rel="noopener">
          this Apple Support article</a> and follow the instructions on "How to change your default shell".
```
minor wording change to make the link text more specific

brownsarahm commented on _includes/swc/setup.html:
```diff
@@ -94,12 +94,16 @@ <h3>The Bash Shell</h3>
-	  Bash by typing <code>bash</code>.
+        </p>
+        <p>
+            In Terminal, check if you have bash. To see if your default shell is Bash
```
```suggestion
             To see if your default shell is Bash
```
Unless the above is untrue Bash is always there, the checking step is only to see if it's the default. 

If it's not always there, then we could add the step to check if it is available in addition to if it's default.

brownsarahm commented:
Overall idea is good, some minor recommendations on the language to make sure it's clear.

fmichonneau committed:
Update _includes/swc/setup.html

Co-authored-by: Sarah Brown <sarah_m_brown@brown.edu>

fmichonneau committed:
Update _includes/swc/setup.html

Co-authored-by: Sarah Brown <sarah_m_brown@brown.edu>

fmichonneau commented:
Thank you both!

fmichonneau committed:
remove extra blank lines
